federation/pkg/federation-controller/util/test/test_helper.go

```diff
@@ -238,6 +239,17 @@ func GetObjectFromChan(c chan runtime.Object) runtime.Object {
 	}
 }
 
+// GetObjectFromChanExpectNone tries to get an api object from the given channel
+// within a short time window where it's expected that there will not be an event.
+func GetObjectFromChanExpectNone(c chan runtime.Object) runtime.Object {
```

Comment:
I think having to use channels to validate a change in operation filtering is excessive.  Consider refactoring ``reconcileConfigMap`` so that generating the list of operations can be validated separately from propagating them.